docs/how_to_guides/how_to_use_ui_api.rst

```diff
@@ -0,0 +1,288 @@
+    for fsi in interface_list:
+        print(f"Got flowsheet: {fsi.display_name}")
+
+
+.. image:: /_static/menu-icon.png
+    :height: 22px
+    :align: left
+
+.. _howto_api-fetch-update:
+
+Fetch and update flowsheet values
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The values for all variables exported by the flowsheet are available via the ``dict()`` method.
+The format of the returned value is documented in the :mod:`~watertap.ui.api` module header.
+Note: if you want to write these values as JSON to a stream, use :meth:`FlowsheetInterface.save`.
+For example::
+
+    from watertap.ui.api import find_flowsheet_interfaces
+    for fsi in find_flowsheet_interfaces():
+        print(f"Got flowsheet: {fsi.display_name}")
+        print(f"Flowsheet contents: {fsi.dict()}")
+
+.. image:: /_static/menu-icon.png
+    :height: 22px
+    :align: left
+
+.. _howto_api-run-actions:
+
+Run flowsheet actions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Once flowsheet actions are created, you invoke them with the :meth:`FlowsheetInterface.run_action` method. The name of the action, if it is a standard one, will be an attribute in the :class:`WorkflowActions` class. For example::
+
+    from watertap.ui.api import FlowsheetInterface
+
+    def run_build(fsi: FlowsheetInterface):
+        fsi.run_action(WorkflowActions.build)
+
+.. image:: /_static/menu-icon.png
+    :height: 22px
+    :align: left
+
+.. _howto_api-save-flowsheet:
+
+Save flowsheet values in a file
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The values of variables that were :func:`exported <export_variables>` for a given flowsheet can be saved to a file.
+For example::
+
+    from watertap.ui.api import FlowsheetInterface
+    fsi = FlowsheetInterface()
+    fsi.set_block(my_flowsheet)
+    fsi.save("my_flowsheet_saved.json")
+
+
+Note: The method is simply a wrapper that calls ``dict()`` and feeds the result to a JSON serializer.
+
+.. image:: /_static/menu-icon.png
+    :height: 22px
+    :align: left
+
+.. _howto_api-load-flowsheet:
+
+Load flowsheet values from a file
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The values of variables that were :ref:`saved <howto_api-save-flowsheet>` for a given flowsheet can be loaded back into the model.
+This operation changes the matching values in the actual underlying model to the values that are stored in the file.
+In other words, unlike the :meth:`FlowsheetInterface.save` method, this method changes what is stored in memory.
+Invoking load is straightforward::
+
+    from watertap.ui.api import FlowsheetInterface
+    # Create and save
+    fsi = FlowsheetInterface()
+    fsi.set_block(my_flowsheet)
+    fsi.save("my_flowsheet_saved.json")
+    # Load back
+    fsi.load("my_flowsheet_saved.json")
+
+To handle situations where the model changes over time, and the saved data and model do not match exactly, there are two methods you can use after you are done loading:
```

Comment:
I meant the former.  Time-dependent models are not something I have tested with, and that is something that should be looked at, since I assumed that indexes are discrete (for starters). But for this section, I have modified the wording to:
```
To handle situations where the model is changed between the save and subsequent load, so that the loaded and current model variables do not all match, there are two methods you can use after you are done loading:

    :meth:`FlowsheetInterface.get_var_missing`
        Returns variables in the loaded data, but not in the current model.

    :meth:`FlowsheetInterface.get_var_extra`
        Returns variables in (and exported by) the current model, but not in the loaded data.

Both methods return a mapping with the full name of the block (e.g., `flowsheet.component.subcomponent`) as the key and a list of variable names as the value.
```
(Apologies for the long lines, I am letting my editor soft-wrap the lines for me and only inserting line breaks for new sentences -- avoids manual reformatting and makes diffs cleaner.)